plugins/engine-datafusion/src/main/java/org/opensearch/datafusion/search/DatafusionReader.java

```diff
@@ -8,18 +8,34 @@
 
 package org.opensearch.datafusion.search;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.LatchedActionListener;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.datafusion.jni.NativeBridge;
 import org.opensearch.datafusion.jni.handle.ReaderHandle;
+import org.opensearch.index.engine.exec.FileStats;
 import org.opensearch.index.engine.exec.WriterFileSet;
 import org.opensearch.index.engine.exec.coord.CatalogSnapshot;
 import org.opensearch.index.engine.exec.coord.CompositeEngine;
 
 import java.io.Closeable;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
 /**
  * DataFusion reader for JNI operations.
  */
 public class DatafusionReader implements Closeable {
+
+    private static final Logger logger = LogManager.getLogger(DatafusionReader.class);
+
+    private static final TimeValue FETCH_TIMEOUT = TimeValue.timeValueMillis(15);
```

Comment:
Seems pretty low, given usual garbage collection pauses are around 20-30ms, can we do 500ms?